### DISCUSSION TITLE
Reference presets from the same repository and revision

### DISCUSSION THREAD

**Discussion by pascal-hofmann** [renovatebot/renovate#25540] [Suggest an Idea]
### Tell us more.

It would be great if renovate would support including presets from the same repository and revision.

This way a big renovate.json with configuration that's only relevant for a single repository could be split into separate files. The description field in these files can then be used to document the intention of the config snippet.

**Comment by HonkingGoose** [comment]
I don't understand _why_ you want to spread your Renovate config over _multiple_ files?

I find it hard enough to remember what I put in the docs repository's `mkdocs.yml` config, what script I need to check my docs-work, and which text I wrote in which Markdown file. 😄 I would _definitely_ get confused by having the config spread out over multiple files. 🙈 

Did you read the [Renovate docs, Shareable config presets](https://docs.renovatebot.com/config-presets/)? It has a section on _Local presets_, that may be what you want?

**Reply by pascal-hofmann** [reply]
Like I wrote, the `renovate.json` in this repo is rather big. It still contains config required only for this repo, so moving into a different repo does not make sense. Also the feedback loop is much shorter during onboarding when files from the same revision can be referenced.

Regarding the confusion when the config is spread out over multiple files: This is the same as with presets, so I don't think this is a valid point. The naming of the files (plus the description shown in the onboarding PR) makes it very clear what is happening.

As for the why:
The separate files can have a descriptive name plus a description inside the json, which document clearly what the config is for (that's what I like most about presets).

**Reply by HonkingGoose** [reply]
Oh okay. I'll let the maintainers respond to your feature request. 🙂

**Reply by rarkins** [reply]
What's an example of how the preset syntax might look for this?

**Reply by pascal-hofmann** [reply]
I have not put much thought into this. I initially thought renovate does support this, and tried with `path/xyz`.

I guess renovate needs some way to distinguish this from other presets. So I guess `//path/xyz` should work? This is similar to the syntax for local presets with a path, just without the `owner/name` part.

**Reply by rarkins** [reply]
It could be more intuitive if the string starts with `./`

**Reply by viceice** [reply]
hard to implement because the current loading of presets. they simply don't know their own path, so they can't compute a relative path to a referenced preset.
